jenkins/Hono-Release-Pipeline.groovy

```diff
@@ -0,0 +1,144 @@
+
+/**
+ * Jenkins pipeline script for Hono release.
+ *
+ */
+
+node {
+    def utils = evaluate readTrusted("jenkins/Hono-PipelineUtils.groovy")
+    properties([buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '3')), parameters([
+            string(defaultValue: '',
+                    description: "The branch to build and release from.\nExamples: \n master\n1.0.x\n1.x",
+                    name: 'BRANCH',
+                    trim: true),
+            string(defaultValue: '',
+                    description: "The version identifier to use for the artifacts built and released by this job. \nExamples:\n1.0.0-M6\n1.0.0-RC1\n2.1.0",
+                    name: 'RELEASE_VERSION',
+                    trim: true),
+            string(defaultValue: '',
+                    description: "The version identifier to use during development of the next version.\nExamples:\n2.0.0-SNAPSHOT\n1.1.0-SNAPSHOT",
+                    name: 'NEXT_VERSION',
+                    trim: true),
+            credentials(credentialType: 'com.cloudbees.plugins.credentials.common.StandardCredentials',
+                    defaultValue: '',
+                    description: 'The credentials to use during checkout from git',
+                    name: 'CREDENTIALS_ID',
+                    required: true)])])
+    try {
+        checkOut()
+        setReleaseVersionAndBuild()
+        commitAndTag()
+        setNextVersion()
+        commitAndPush()
+        copyArtifacts()
+        utils.publishJavaDoc('target/site/apidocs', true)
+        currentBuild.result = 'SUCCESS'
+    } catch (err) {
+        currentBuild.result = 'FAILURE'
+        echo "Error: ${err}"
+    }
+    finally {
+        echo "Build status: ${currentBuild.result}"
+        utils.notifyBuildStatus()
+    }
+}
+
+/**
+ * Checks out the specified branch from git repo
+ *
+ */
+def checkOut() {
+    stage('Checkout') {
+        echo "Check out branch: origin/${params.BRANCH}"
+        checkout([$class                           : 'GitSCM',
+                  branches                         : [[name: "origin/${params.BRANCH}"]],
+                  doGenerateSubmoduleConfigurations: false,
+                  extensions                       : [[$class: 'WipeWorkspace'],
+                                                      [$class: 'LocalBranch']],
+                  userRemoteConfigs                : [[credentialsId: "${params.CREDENTIALS_ID}", url: 'ssh://github.com/eclipse/hono.git']]])
+    }
+}
+
+/**
+ * Set version to RELEASE_VERSION and build using maven
+ *
+ */
+def setReleaseVersionAndBuild() {
+    stage('Build') {
+        withMaven(maven: 'apache-maven-latest', jdk: 'jdk1.8.0-latest', options: [jacocoPublisher(disabled: true), artifactsPublisher(disabled: true)]) {
+            sh "mvn versions:set -DgenerateBackupPoms=false -DnewVersion=${params.RELEASE_VERSION}"
```

Comment:
shouldn't this be `${RELEASE_VERSION}` as well? Does it matter at all, i.e. do both versions work?

Reply:
In this place it works fine. Today I tested with various scenarios and I found issues using ${params.XXX} inside multi-line shell scripts (sh ''' <content> '''). I would like to keep it uniform  throughout all shell script invocations, I will change ${params.RELEASE_VERSION} to ${RELEASE_VERSION}